### PR TITLE
Allow tags on PSR-6

### PR DESCRIPTION
**Pull request by Nyholm**
Im really impressed by this library. Really good job!

Here is a PR that allow your PSR-6 implementation to support tagging. It uses a trait and interface from https://github.com/php-cache/taggable-cache

(I am the author of the trait)

### CODE REVIEW

**Comment by Nyholm** [comment]
I changed the title to WIP. 

I'm going to make sure that the existing key are not updated.

File: composer.json [hunk 1]
```diff
@@ -27,12 +27,14 @@
     }
   ],
   "require": {
-    "php": ">=5.3.0",
-    "psr/cache": "~1.0"
+    "php": ">=5.4",
```
Comment (cryptiklemur) [inline composer.json]: ^5.4|^7.0

Comment (cryptiklemur) [inline composer.json]: should change them all to ^ actually

Comment (Nyholm) [inline composer.json]: > ^5.4|^7.0

Yeah, you are correct. Because we do not know if this will work in php 8.0

**Comment by Nyholm** [comment]
I also added some integration tests to this PR. There are some minor issues with the implementation. These issues existed before I made this PR. This PR does not break/change current behavior. 

I am ready to merge.

**Comment by matthiasmullie** [comment]
I'll get back to this in detail in a couple of days!

**Comment by matthiasmullie** [comment]
Alright. This particular pull request didn't feel exactly right for how this project is currently built: there are a bunch of layer that can just wrap around the previous.
I thought this taggable cache could also just wrap wround the default PSR-6 implementation, and decided to make it just like that. Eventually, I started to notice that it could just wrap around any PSR-6 implementation.

Check it out at https://github.com/matthiasmullie/taggable-cache & please let me know any feedback!

Also, 1 minor change would be required to taggable-cache to actually make it work: https://github.com/php-cache/taggable-cache/pull/15

**Comment by Nyholm** [comment]
I see. It was foolish of me to mix the PRs. Ive should have made one for integration tests and one for the tagging. 

I like your tagging cache pool. It looks good. I think I saw some minor issues with it. Activate the integration tests and you will see whats wrong. I can sen a PR to this repo.

**Comment by matthiasmullie** [comment]
Done. I decided not to split it into a separate project after all (because it would probably just end up not being used, separately)

The code is now included in the main Scrapbook repo: https://github.com/matthiasmullie/scrapbook/tree/master/src/Psr6/Taggable (also see http://www.scrapbook.cash/projects/taggable-cache.html)

It should wrap around any valid PSR-6 implementation.

**Comment by Nyholm** [comment]
Nice!

**Pull request closed without merging**